mycore-ocfl/src/main/java/org/mycore/ocfl/metadata/MCROCFLXMLMetadataManager.java

```diff
@@ -317,7 +329,60 @@ public IntStream getStoredIDs(String project, String type) throws MCRPersistence
 
     @Override
     public int getHighestStoredID(String project, String type) {
-        return getStoredIDs(project, type).max().orElse(0);
+        Path path
+            = Path.of(MCRConfiguration2.getStringOrThrow("MCR.OCFL.Repository." + repositoryKey + ".RepositoryRoot"));
+        int highestStoredID = 0;
+        int maxDepth = Integer.MAX_VALUE;
+        OcflExtensionConfig config = MCRConfiguration2.getSingleInstanceOf("MCR.OCFL.Repository." + repositoryKey)
```

Comment:
Will not work with S3

Reply:
The default Repository Providers don't allow the use of S3, since they are all fileSystem storage.
```java
this.repository = new OcflRepositoryBuilder()
            .defaultLayoutConfig(getExtensionConfig())
            .storage(storage -> storage.fileSystem(repositoryRoot))
            .workDir(workDir)
            .build();```
https://github.com/OCFL/ocfl-java/blob/main/docs/USAGE.md#amazon-s3
When creating an S3 RepositoryProvider, the MetadataManagers method can also be replaced for the S3 store.